bundles/org.eclipse.ui.workbench/META-INF/MANIFEST.MF

```diff
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: %pluginName
 Bundle-SymbolicName: org.eclipse.ui.workbench; singleton:=true
-Bundle-Version: 3.132.0.qualifier
+Bundle-Version: 4.0.0.qualifier
```

Comment:
This is a no go.  We can’t have broken api.

Reply:
It is broken in master branch. See the commit about MultiPageEditor.

Reply:
I'm investigating why master changes affect PR build's API checks in https://github.com/eclipse-platform/eclipse.platform.ui/pull/1254

Reply:
Confirmed, a PR without any changes at all it failing due to previously broken API lacking version bump.

https://github.com/eclipse-platform/eclipse.platform.ui/pull/1254
https://github.com/eclipse-platform/eclipse.platform.ui/actions/runs/6677970750/job/18148533353?pr=1254

Reply:
Very strange indeed. 😱

Reply:
If you look closer at the log you will see:

```
14:36:20.557 [WARNING] [API WARNING] File MANIFEST.MF at line 0: The API problem filter for: 'The major version should be incremented in version 3.131.0, since API breakage occurred since version 3.130.0' is no longer used (location: /home/jenkins/agent/workspace/eclipse.platform.ui_PR-1253/bundles/org.eclipse.ui.workbench/META-INF/MANIFEST.MF)
14:36:20.557 [WARNING] [API WARNING] File MANIFEST.MF at line 0: The API problem filter for: 'The type org.eclipse.ui.part.MultiPageEditor has been removed from org.eclipse.ui.workbench_3.131.0' is no longer used (location: /home/jenkins/agent/workspace/eclipse.platform.ui_PR-1253/bundles/org.eclipse.ui.workbench/META-INF/MANIFEST.MF)
```

so the API filter needs to be adjusted due to the changed version of the manifest

Reply:
The argument here was, that verison should not have been changed at all. This PR does not have any API breakages.

Reply:
Replaced major version bump with minor version bump and added another filter.

Reply:
I've failed to bump minor version. If I do, IDE complains, that the bump is unnecessary. I suspect this is caused by the problem filter.

Reply:
Yes this is a nightmare, you need then another filter for the bump as well ...

Reply:
> Yes this is a nightmare, you need then another filter for the bump as well ...

Adding filter makes version bump unnecessary
Removing filter reports API breakage for the bumped version

There is no way to bump version here anymore.

Reply:
You need one filter that suppress the major version bump, then you increase the minor and add another filter that suppress the "unnecessary" version bump error.

Reply:
I thought @laeubi suggested the filter needs updating to be applicable for the updated version?

Reply:
Yes, when filter is updated, IDE complains, that the bump is unnecessary.

Reply:
Why do you think you need a version bump at all?

Reply:
@iloveeclipse internal API in package `org.eclipse.ui.internal.handlers` is extended. It is shared with `org.eclipse.ui` and `org.eclipse.ui.ide`. Also, Jenkins complains if no version bump happens when code and qualifier are changed, but not version.

Reply:
Anyway, I've removed the version bump (which should not be done, as API and code have changed) and the API verification is now passing. Maybe  @akurtakov can give some advice later, how to fix this.
Meanwhile, Jenkins produces test results so I can continue working on this draft.

Thanks for the help everyone!

Reply:
There is no "internal" API  for API  tooling. So if **real** API isn't changed, **and** bundle version changed already during 4.30 cycle, no bump is needed.

Reply:
> There is no "internal" API for API tooling.
@iloveeclipse thanks, I did not know that.
Exported package:
```
org.eclipse.ui.internal.handlers;x-friends:="org.eclipse.ui,org.eclipse.ui.ide",
```
looked "API enough" to me :sweat_smile: